gnucash/import-export/import-pending-matches.cpp

```diff
@@ -70,28 +67,40 @@ gnc_import_PendingMatches_remove_match (GNCImportPendingMatches& map,
 {
     g_return_if_fail (match_info);
 
-    auto& pending_matches = gnc_import_PendingMatches_get_value (map, match_info);
+    auto match_guid = gnc_import_PendingMatches_get_key (match_info);
+
+    auto iter = map.find (match_guid);
+    if (iter == map.end())
+        throw std::out_of_range("coding error. trying to remove missing match");
+
+    auto& pending_matches = iter->second;
 
     if (selected_manually)
         pending_matches.num_manual_matches--;
     else
         pending_matches.num_auto_matches--;
+
+    if (pending_matches.all_zero())
+        map.erase (iter);
 }
 
 GNCImportPendingMatchType
-gnc_import_PendingMatches_get_match_type (GNCImportPendingMatches& map,
+gnc_import_PendingMatches_get_match_type (const GNCImportPendingMatches& map,
                                           GNCImportMatchInfo *match_info)
 {
     g_return_val_if_fail (match_info, GNCImportPending_NONE);
 
-    auto& pending_matches = gnc_import_PendingMatches_get_value (map, match_info);
-
-    if (pending_matches.num_manual_matches > 0)
-        return GNCImportPending_MANUAL;
-
-    if (pending_matches.num_auto_matches > 0)
-        return GNCImportPending_AUTO;
+    auto match_guid = gnc_import_PendingMatches_get_key (match_info);
+    auto iter = map.find (match_guid);
+    if (iter != map.end())
+    {
+        const auto& pending_matches = iter->second;
+        if (pending_matches.num_manual_matches > 0)
+            return GNCImportPending_MANUAL;
 
+        if (pending_matches.num_auto_matches > 0)
+            return GNCImportPending_AUTO;
+    }
     return GNCImportPending_NONE;
```

Comment:
Hmm. Is it supposed to be possible for a PendingMatches to have both manual and auto values?

Reply:
Yes but historically manual>0 always had priority...